### ISSUE TITLE
[Bug] Profling Python Stack Trace is Error when run evalscope and capture profiling

### ISSUE THREAD

**Issue by pengchengneo** [sgl-project/sglang-jax#473]
### Checklist

- [x] 1. I have searched related issues but cannot get the expected help.
- [x] 2. The bug has not been fixed in the latest version.
- [x] 3. Please note that if the bug-related issue you submitted lacks corresponding environment info and a minimal reproducible demo, it will be challenging for us to reproduce and resolve the issue, reducing the likelihood of receiving feedback.
- [x] 4. If the issue you raised is not a bug but a question, please raise a discussion at https://github.com/sgl-project/sgl-jax/discussions/new/choose Otherwise, it will be closed.
- [x] 5. Please use English, otherwise it will be closed.

### Describe the bug

`python3 -u -m sgl_jax.launch_server   --model-path /models/Qwen/Qwen3-32B   --trust-remote-code   --device=tpu    --mem-fraction-static=0.8   --max-prefill-tokens=4096   --max-running-requests=16  --decode-log-interval=1   --enable-request-time-stats-logging   --attention-backend=fa   --dtype=bfloat16  --port 30000 --host 0.0.0.0 --disable-overlap-schedule --tp-size 4 --speculative-algorithm EAGLE3 --speculative-draft-model-path  AngelSlim/Qwen3-32B_eagle3 --page-size 64 --speculative-eagle-topk 1 --speculative-num-steps 3 --speculative-num-draft-tokens 4 --speculative-draft-model-revision 67caf31f9062d7ab64872e0a111d499bc16cd205`

` evalscope eval  --model /models/Qwen/Qwen3-32B --api-url http://127.0.0.1:30000/v1/chat/completions --datasets gsm8k --eval-batch-size 8 --eval-type service --limit 16`

### Reproduction

<img width="3840" height="1918" alt="Image" src="https://github.com/user-attachments/assets/35d97b22-b2db-4b9c-a316-d5cb5f530135" />

stack  trace is wrong, misalignment with the true python trace

### Environment

`Package                  Version       Editable project location
------------------------ ------------- ----------------------------
absl-py                  2.3.1
accelerate               1.11.0
addict                   2.4.0
aiofiles                 25.1.0
aiohappyeyeballs         2.6.1
aiohttp                  3.13.2
aiosignal                1.4.0
annotated-types          0.7.0
antlr4-python3-runtime   4.13.2
anyio                    4.11.0
attrs                    25.4.0
certifi                  2025.11.12
charset-normalizer       3.4.4
chex                     0.1.90
click                    8.3.1
colorama                 0.4.6
contourpy                1.3.3
cycler                   0.12.1
datasets                 3.2.0
dill                     0.3.8
distro                   1.9.0
einops                   0.8.1
etils                    1.13.0
evalscope                0.17.1
fastapi                  0.116.2
filelock                 3.20.0
flax                     0.12.1
fonttools                4.60.1
frozenlist               1.8.0
fsspec                   2024.9.0
h11                      0.16.0
hf-xet                   1.2.0
httpcore                 1.0.9
httpx                    0.28.1
huggingface-hub          0.34.6
humanize                 4.14.0
idna                     3.11
immutabledict            4.2.2
importlib_resources      6.5.2
jax                      0.8.1
jaxlib                   0.8.1
jieba                    0.42.1
Jinja2                   3.1.6
jiter                    0.12.0
joblib                   1.5.2
jsonlines                4.0.0
kiwisolver               1.4.9
langdetect               1.0.9
latex2sympy2_extended    1.10.2
libtpu                   0.0.30
llguidance               1.3.0
lxml                     6.0.2
markdown-it-py           4.0.0
MarkupSafe               3.0.3
matplotlib               3.10.7
mdurl                    0.1.2
ml_dtypes                0.5.4
modelscope               1.32.0
mpmath                   1.3.0
msgpack                  1.1.2
msgpack-python           0.5.6
multidict                6.7.0
multiprocess             0.70.16
nest-asyncio             1.6.0
networkx                 3.5
nltk                     3.9.2
numpy                    2.3.5
nvidia-cublas-cu12       12.8.4.1
nvidia-cuda-cupti-cu12   12.8.90
nvidia-cuda-nvrtc-cu12   12.8.93
nvidia-cuda-runtime-cu12 12.8.90
nvidia-cudnn-cu12        9.10.2.21
nvidia-cufft-cu12        11.3.3.83
nvidia-cufile-cu12       1.13.1.3
nvidia-curand-cu12       10.3.9.90
nvidia-cusolver-cu12     11.7.3.90
nvidia-cusparse-cu12     12.5.8.93
nvidia-cusparselt-cu12   0.7.1
nvidia-nccl-cu12         2.27.5
nvidia-nvjitlink-cu12    12.8.93
nvidia-nvshmem-cu12      3.3.20
nvidia-nvtx-cu12         12.8.90
openai                   2.8.1
opt_einsum               3.4.0
optax                    0.2.6
orbax-checkpoint         0.11.28
orjson                   3.11.4
packaging                25.0
pandas                   2.3.3
partial-json-parser      0.2.1.1.post7
pillow                   11.3.0
pip                      25.3
portalocker              3.2.0
propcache                0.4.1
protobuf                 6.33.1
psutil                   7.1.3
pyarrow                  22.0.0
pybase64                 1.4.2
pydantic                 2.12.4
pydantic_core            2.41.5
Pygments                 2.19.2
pyparsing                3.2.5
python-dateutil          2.9.0.post0
python-multipart         0.0.20
pytz                     2025.2
PyYAML                   6.0.3
pyzmq                    27.0.2
regex                    2025.11.3
requests                 2.32.5
rich                     14.2.0
rouge-chinese            1.0.3
rouge_score              0.1.2
sacrebleu                2.5.1
safetensors              0.7.0
scikit-learn             1.7.2
scipy                    1.16.3
seaborn                  0.13.2
setproctitle             1.3.7
setuptools               80.9.0
sglang-jax               0.0.2         /home/gcpuser/sgl-jax/python
simplejson               3.20.2
six                      1.17.0
sniffio                  1.3.1
sortedcontainers         2.4.0
starlette                0.48.0
sympy                    1.14.0
tabulate                 0.9.0
tensorstore              0.1.79
threadpoolctl            3.6.0
tiktoken                 0.12.0
tokenizers               0.22.1
toolz                    1.1.0
torch                    2.9.1
tqdm                     4.67.1
transformers             4.57.1
treescope                0.1.10
triton                   3.5.1
typing_extensions        4.15.0
typing-inspection        0.4.2
tzdata                   2025.2
urllib3                  2.5.0
uvicorn                  0.35.0
uvloop                   0.21.0
wheel                    0.45.1
word2number              1.1
xxhash                   3.6.0
yarl                     1.22.0
zipp                     3.23.0`

**Comment by SiqiLi-Fighting** [comment]
Solution: Add `with jax.profiler.TraceAnnotation` where the `with` operation is needed to enforce scope isolation

**Comment by SiqiLi-Fighting** [comment]
```
import jax
import jax.numpy as jnp
from jax.sharding import Mesh, PartitionSpec, NamedSharding
from jax.experimental import mesh_utils
import time
import os

device_mesh = mesh_utils.create_device_mesh((4,))
mesh = Mesh(device_mesh, axis_names=('x',))

@jax.jit
def heavy_op(x):
    return x @ x

sharding = NamedSharding(mesh, PartitionSpec('x', None))
data = jax.device_put(jnp.ones((4096, 4096)), sharding)
heavy_op(data).block_until_ready()

def run_test(name, use_mesh_inside_loop):
    trace_dir = f"/tmp/profiling/jax_trace_{name}"
    print(f"Running {name}... Trace will be saved to {trace_dir}")
    
    jax.profiler.start_trace(trace_dir)
    
    if use_mesh_inside_loop:
        for _ in range(20):
            with jax.sharding.set_mesh(mesh):
                heavy_op(data).block_until_ready()
    else:
        with jax.sharding.set_mesh(mesh):
            for _ in range(20):
                heavy_op(data).block_until_ready()
                
    jax.profiler.stop_trace()
    print(f"Done {name}.")

if __name__ == "__main__":
    run_test("bad_stacking", use_mesh_inside_loop=True)
    time.sleep(1) 
    run_test("normal_flat", use_mesh_inside_loop=False)

```

I tested this error with above script, and found this error is imported by "with" operation, when use "with" operation , CPU side with scope and device side scope is not aligned on timeline, make CPU stach is abnormal , and the fix solution is that we add "jax.profiler.TraceAnnotation" when we use "with" operation to avoid this error
